controllers/profile_controller.go

```diff
@@ -102,27 +103,22 @@ func (r *ProfileSubscriptionReconciler) Reconcile(ctx context.Context, req ctrl.
 		return ctrl.Result{}, err
 	}
 
-	instance := profile.New(pDef, pSub, r.Client, logger)
-	artifactStatus, err := instance.ArtifactStatus(ctx)
+	instance := profile.New(ctx, pDef, pSub, r.Client, logger)
+
+	err = instance.ReconcileArtifacts()
```

Comment:
Would you mind inlining this in the iff below with `if err := ..; err != nil`? :)